### PR TITLE
Manually add compute_instance

### PR DESCRIPTION
**Pull request by nstogner**
Add google_compute_instance manually. Steps to reproduce (from magic modules repo):

```sh
# Edit products/compute/terraform.yaml and set "exclude: false" for compute instance
git clone github.com/GoogleCloudPlatform/terraform-google-conversion $GOPATH/src/github.com/GoogleCloudPlatform/terraform-google-conversion
bundle exec compiler --product products/compute/ --type Instance -e terraform -f validator -o "$GOPATH/src/github.com/GoogleCloudPlatform/terraform-google-conversion"
```

### CODE REVIEW

File: google/compute_instance.go [hunk 1]
```diff
@@ -0,0 +1,764 @@
+// ----------------------------------------------------------------------------
```
Comment (ndmckinley) [inline google/compute_instance.go]: Ideal to remove this header since it is not, in practice, autogenerated.

File: google/compute_instance.go [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+
+func GetComputeInstanceApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+	obj := make(map[string]interface{})
+	canIpForwardProp, err := expandComputeInstanceCanIpForward(d.Get("can_ip_forward"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("can_ip_forward"); !isEmptyValue(reflect.ValueOf(canIpForwardProp)) && (ok || !reflect.DeepEqual(v, canIpForwardProp)) {
+		obj["canIpForward"] = canIpForwardProp
+	}
+	disksProp, err := expandComputeInstanceDisks(d.Get("disks"), d, config)
```
Comment (ndmckinley) [inline google/compute_instance.go]: This is quite a bit more complicated in practice - needs to handle boot vs scratch disks vs attached disks.

File: google/compute_instance.go [hunk 3]
```diff
@@ -0,0 +1,764 @@
...
+	} else if v, ok := d.GetOkExists("min_cpu_platform"); !isEmptyValue(reflect.ValueOf(minCpuPlatformProp)) && (ok || !reflect.DeepEqual(v, minCpuPlatformProp)) {
+		obj["minCpuPlatform"] = minCpuPlatformProp
+	}
+	nameProp, err := expandComputeInstanceName(d.Get("name"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("name"); !isEmptyValue(reflect.ValueOf(nameProp)) && (ok || !reflect.DeepEqual(v, nameProp)) {
+		obj["name"] = nameProp
+	}
+	networkInterfacesProp, err := expandComputeInstanceNetworkInterfaces(d.Get("network_interfaces"), d, config)
```
Comment (ndmckinley) [inline google/compute_instance.go]: This block is actually named "network_interface", singular.

File: google/compute_instance.go [hunk 4]
```diff
@@ -0,0 +1,764 @@
...
+	schedulingProp, err := expandComputeInstanceScheduling(d.Get("scheduling"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("scheduling"); !isEmptyValue(reflect.ValueOf(schedulingProp)) && (ok || !reflect.DeepEqual(v, schedulingProp)) {
+		obj["scheduling"] = schedulingProp
+	}
+	serviceAccountsProp, err := expandComputeInstanceServiceAccounts(d.Get("service_accounts"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("service_accounts"); !isEmptyValue(reflect.ValueOf(serviceAccountsProp)) && (ok || !reflect.DeepEqual(v, serviceAccountsProp)) {
```
Comment (ndmckinley) [inline google/compute_instance.go]: This name is also singular - `service_account`.

File: google/compute_instance.go [hunk 5]
```diff
@@ -0,0 +1,764 @@
...
+	} else if v, ok := d.GetOkExists("tags"); !isEmptyValue(reflect.ValueOf(tagsProp)) && (ok || !reflect.DeepEqual(v, tagsProp)) {
+		obj["tags"] = tagsProp
+	}
+	zoneProp, err := expandComputeInstanceZone(d.Get("zone"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("zone"); !isEmptyValue(reflect.ValueOf(zoneProp)) && (ok || !reflect.DeepEqual(v, zoneProp)) {
+		obj["zone"] = zoneProp
+	}
+
```
Comment (ndmckinley) [inline google/compute_instance.go]: Missing:
- description
- deletion_protection
- hostname
- metadata_startup_script

File: google/compute_instance.go (hunk 1 above)
Comment (morgante) [reply google/compute_instance.go]: This part confused me - I thought we'd end up landing these manual resources upstream in magic-modules not by writing the files directly here?

Comment (ndmckinley) [reply google/compute_instance.go]: Either approach is acceptable!  We used the "mm-only" model for our other terraform stuff, but we don't have to.  :)

Comment (morgante) [reply google/compute_instance.go]: I'd think MM would be more consistent and it *looks* like @nstogner actually generated this via MM. Probably the right thing is to start a PR on the MM repo instead?

Comment (ndmckinley) [reply google/compute_instance.go]: I'm open to either approach.  If you add it to MM, go ahead and put the file itself in `third_party`, and add it to `provider/terraform_object_library.rb`'s `copy_common_files` list.  Parallel to, e.g. `cai.go`.

Comment (nstogner) [reply google/compute_instance.go]: Ok, so if we make the PR to MM, the process is pretty much the same: generate the resource if the templates change, manually patch it, and add to `third_party/` ?

Comment (nstogner) [reply google/compute_instance.go]: Thanks, I will take that approach.

Comment (ndmckinley) [reply google/compute_instance.go]: :+1: - closed this one.

**Pull request closed without merging**